controllers/provisioning_controller.go

```diff
@@ -445,18 +440,23 @@ func (r *ProvisioningReconciler) apiServerInternalHost(ctx context.Context) (str
 	return host, nil
 }
 
-func (r *ProvisioningReconciler) masterMacAddresses(ctx context.Context) ([]string, error) {
+func (r *ProvisioningReconciler) updateProvisioningMacAddresses(ctx context.Context, provConfig *metal3iov1alpha1.Provisioning) error {
+	if len(provConfig.Spec.ProvisioningMacAddresses) != 0 {
+		return nil
+	}
+
 	macs := []string{}
 	bmhl := baremetalv1alpha1.BareMetalHostList{}
 	if err := r.Client.List(ctx, &bmhl, &client.ListOptions{Namespace: ComponentNamespace}); err != nil {
-		return nil, err
+		return err
 	}
 	for _, bmh := range bmhl.Items {
 		if strings.Contains(bmh.Name, "master") && len(bmh.Spec.BootMACAddress) > 0 {
 			macs = append(macs, bmh.Spec.BootMACAddress)
 		}
 	}
-	return macs, nil
+	provConfig.Spec.ProvisioningMacAddresses = macs
+	return r.Client.Update(ctx, provConfig)
```

Comment:
we can later move this to the installer

Reply:
Yeah I was wondering if we should just calculate this in the installer, and CBO can then just consume the provisioning CR value like other install-time-provided things.

That also partially resolves the issue that bootMACAddress of the BMH is optional, but in the install-config it's mandatory

Reply:
Will it always be a one-time config to happen at install time? What if one of the master nodes gets replaced, should this list of bootMacAddresses have get recalculated?